hand-eye-calibration/full_hand_eye_sample.py

```diff
@@ -0,0 +1,539 @@
+"""
+Script to generate a dataset and perform hand-eye calibration using a Universal Robot UR5e robot.
+The script communicates with the robot through TCP/IP. The host is defined by the user.
+
+The entire sample consist of two additional files:
+    - universal_robots_hand_eye_script.urp: Robot program script that moves between different poses.
+    - robot_communication_file.xml: communication set-up file.
+
+Running the sample requires that you have universal_robots_hand_eye_script.urp on your UR5e robot,
+and robot_communication_file.xml in the same repo as this sample.
+Each robot pose must be modified to your scene. This is done in universal_robots_hand_eye_script.urp on the robot.
+
+Further explanation of this sample is found in our knowledge base:
+Further explaination of this sample is found in our knowledge base:
+https://zivid.atlassian.net/wiki/spaces/ZividKB/pages/103481384/
+"""
+from pathlib import Path
+import time
+import datetime
+
+import cv2
+import numpy as np
+import zivid
+import rtde.rtde as rtde
+import rtde.rtde_config as rtde_config
+
+
+def quat_to_rotm(quat: np.array) -> np.array:
+    """Convert from quaternion to rotation matrix
+
+    Args:
+        quat: Rotations in quaternions
+
+    Returns:
+        Rotations in degrees as 3x3 rotation matrix
+    """
+
+    q_w = quat[0]
+    q_x = quat[1]
+    q_y = quat[2]
+    q_z = quat[3]
+
+    rot_matrix = np.zeros((3, 3))
+    rot_matrix[0, 0] = 1 - 2 * np.power(q_y, 2) - 2 * np.power(q_z, 2)
+    rot_matrix[0, 1] = 2 * q_x * q_y - 2 * q_z * q_w
+    rot_matrix[0, 2] = 2 * q_x * q_z + 2 * q_y * q_w
+    rot_matrix[1, 0] = 2 * q_x * q_y + 2 * q_z * q_w
+    rot_matrix[1, 1] = 1 - 2 * np.power(q_x, 2) - 2 * np.power(q_z, 2)
+    rot_matrix[1, 2] = 2 * q_y * q_z - 2 * q_x * q_w
+    rot_matrix[2, 0] = 2 * q_x * q_z - 2 * q_y * q_w
+    rot_matrix[2, 1] = 2 * q_y * q_z + 2 * q_x * q_w
+    rot_matrix[2, 2] = 1 - 2 * np.power(q_x, 2) - 2 * np.power(q_y, 2)
+
+    return rot_matrix
+
+
+def rotvec_to_quat(rotvec: np.array, angle: float = None) -> np.array:
```

Comment:
It seems that this is angle-axis, not rotation vector? Or am I wrong and it can handle both?

Reply:
done. using scipy isntead